src/main/java/org/dice_group/grp/grammar/digram/Digram.java

```diff
@@ -82,4 +78,51 @@ public void setNoOfOccurences(long occurences) {
 		this.occurences = occurences;
 	}
 	
+
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((edgeLabel1 == null) ? 0 : edgeLabel1.hashCode());
+		result = prime * result + ((edgeLabel2 == null) ? 0 : edgeLabel2.hashCode());
+		result = prime * result + ((external == null) ? 0 : external.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
```

Comment:
instead you can ask if(obj instanceof Digram) 
In this case it does not matter though